### PR TITLE
fix(android/browser): Improve how embedded browser handles input

### PR DESCRIPTION
**Pull request by darcywong00**
While checking Sentry logs for #3721, I noticed a large amount of [MalformedURLExceptions](http://sentry.keyman.com/organizations/keyman/issues/484/?project=7&query=is%3Aunresolved).

(edited)
This PR adds the following fixes to `WebBrowserActivity`:
* Don't log `MalformedURLExceptions` since it may be a query
* URLEncode the string if it seems to be a URL
* Set `android:usesCleartextTraffic="true"` which allows http traffic (API 23+)

### CODE REVIEW

File: android/KMAPro/kMAPro/src/main/java/com/tavultesoft/kmapro/WebBrowserActivity.java [hunk 1]
```diff
@@ -323,7 +323,10 @@ public boolean onEditorAction(TextView v, int actionId, KeyEvent event) {
           try {
             new URL(urlStr);
           } catch (MalformedURLException e) {
-            KMLog.LogException(TAG, "", e);
+            // Intentionally not logging "No protocol" exception because it may be a query
+            if (e != null && !e.getMessage().startsWith("no protocol:")) {
+              KMLog.LogException(TAG, "", e);
```
Comment (mcdurdin) [inline android/KMAPro/kMAPro/src/main/java/com/tavultesoft/kmapro/WebBrowserActivity.java:328]: This should never be logged as an exception (`KMLog.d()` would be okay) -- no matter what the format. We don't want exceptions logged for user input, only for errors in code.

File: android/KMAPro/kMAPro/src/main/java/com/tavultesoft/kmapro/WebBrowserActivity.java [hunk 2]
```diff
@@ -323,7 +323,10 @@ public boolean onEditorAction(TextView v, int actionId, KeyEvent event) {
...
+            }
 
             if (Patterns.WEB_URL.matcher(String.format("%s%s", "http://", urlStr)).matches()) {
```
Comment (mcdurdin) [inline android/KMAPro/kMAPro/src/main/java/com/tavultesoft/kmapro/WebBrowserActivity.java:331]: This subsequent test is inadequate: only treat as URL if throwing `http://` on the front makes it look like a URL according to our `WEB_URL` pattern? That may still throw if `webView` thinks the resulting URL is still invalid. It would be better to loop around `new URL(urlStr)` after modifying `urlStr` to add `http://` on the front, i.e. something like this:

```java
bool valid = false;
string originalUrl = urlStr;
while(!valid) {
  try {
    new URL(urlStr);
    valid = true;
  } catch (MalformedURLException e) {
    if (Patterns.WEB_URL.matcher("http://" + urlStr).matches()) {
      urlStr = "http://" + originalUrl;
    } else {
      urlStr = String.format("https://www.google.com/search?q=%s", encodeUriComponent(originalUrl));
      valid = true; // no need to test again; we'll assume it's valid this time
    }
  }
}
```

* Where `encodeUriComponent` is the Java equivalent (which I don't know off the top of my head!) -- this was a bug before.
* Note not using `String.format` (overkill!) for simple string concatenation

**Review by mcdurdin — CHANGES_REQUESTED** [review]

File: android/KMAPro/kMAPro/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -15,6 +15,7 @@
     android:icon="@drawable/ic_launcher"
     android:label="@string/app_name"
     android:roundIcon="@drawable/ic_launcher_round"
+    android:usesCleartextTraffic="true"
```
Comment (mcdurdin) [inline android/KMAPro/kMAPro/src/main/AndroidManifest.xml:18]: This should only be true for debug builds? Why did you need to add it now?

File: android/KMAPro/kMAPro/src/main/java/com/tavultesoft/kmapro/WebBrowserActivity.java [hunk 4]
```diff
@@ -320,18 +323,31 @@ public boolean onEditorAction(TextView v, int actionId, KeyEvent event) {
...
+          String originalUrl = urlStr;
+          while(!valid) {
+            try {
+              new URL(urlStr);
+              valid = true;
+            } catch (MalformedURLException e) {
+              // Intentionally not logging exception because it may be a query
+
+              if (Patterns.WEB_URL.matcher("http://" + urlStr).matches()) {
+                urlStr = "http://" + originalUrl;
```
Comment (mcdurdin) [inline android/KMAPro/kMAPro/src/main/java/com/tavultesoft/kmapro/WebBrowserActivity.java:337]:
```suggestion
              if (Patterns.WEB_URL.matcher("http://" + urlStr).matches()) {
                urlStr = "http://" + urlStr;
```

File: android/KMAPro/kMAPro/src/main/AndroidManifest.xml (hunk 3 above)
Comment (darcywong00) [reply android/KMAPro/kMAPro/src/main/AndroidManifest.xml:18]: The embedded browser was blocking regular http:// traffic with the error
```
net::ERR_CLEARTEXT_NOT_PERMITTED
```

**Review by mcdurdin — APPROVED** [review]
